benchopt/runner.py

```diff
@@ -205,6 +205,36 @@ def get_next(x): return max(x / rho, MIN_TOL)
     return curve, status
 
 
+def get_callback(curve_one_rep, cb_status, objective, solver,
+                 meta_rep, deadline):
+    # make documentation
+    info = get_sys_info()
+
+    def callback(it, beta_hat_i):
+        t0 = time.perf_counter()
+        cb_status["delta_t"] += t0 - cb_status["time"]
+        stop = False
+        if it == cb_status["next_stopval"]:
+            objective_dict = objective(beta_hat_i)
+            # Add system info in results
+            dict_ = dict(**meta_rep, solver_name=str(solver),
```

Comment:
Solver should be added in `meta_rep`, this will make it easier to read the code.

Reply:
I made both changes in the last commit :)